src/librekey/rnp_key_store.cpp

```diff
@@ -58,45 +58,70 @@
 #include "pgp-key.h"
 #include "fingerprint.h"
 #include "crypto/hash.h"
+#include "file-utils.h"
+#ifdef _WIN32
+#include <locale>
+#include <codecvt>
+#endif
 
 bool
 rnp_key_store_load_from_path(rnp_key_store_t *         key_store,
                              const pgp_key_provider_t *key_provider)
 {
-    DIR *          dir;
-    bool           rc;
-    pgp_source_t   src = {};
+    bool         rc;
+    pgp_source_t src = {};
+#ifdef _WIN32
+    struct _wdirent *ent;
+#else
     struct dirent *ent;
-    char           path[MAXPATHLEN];
+#endif
 
     if (key_store->format == PGP_KEY_STORE_G10) {
-        dir = opendir(key_store->path.c_str());
+        auto dir = rnp_opendir(key_store->path.c_str());
         if (dir == NULL) {
             RNP_LOG(
               "Can't open G10 directory %s: %s", key_store->path.c_str(), strerror(errno));
             return false;
         }
 
-        while ((ent = readdir(dir)) != NULL) {
-            if (!strcmp(ent->d_name, ".") || !strcmp(ent->d_name, "..")) {
+        while ((ent = rnp_readdir(dir)) != NULL) {
```

Comment:
As we make use only of ent->d_name here, what about moving most of this code to the function like `std::string rnp_readdir_name()`, and getting read of numerous ifdefs here, string conversions and so on?

Reply:
this is done to avoid unnecessary conversion from wide char to utf-8 because `_wreaddir` returns wide char, and we need to open the file with `_wopen`, let it pass directly to `init_fle_src_w`

Reply:
Wide char packing to utf-8 and unpacking is of linear complexity, and given that it would work on file names, i.e. short strings, performance benefits would be neglected with the more complex code/more code lines.